pkg/pool-utils/contracts/ProtocolFeeCache.sol

```diff
@@ -126,6 +128,13 @@ abstract contract ProtocolFeeCache is RecoveryMode {
         _updateProtocolFeeCache(_protocolFeeProvider, ProtocolFeeType.AUM);
     }
 
+    /**
+     * @dev Override in derived contracts to perform some action before the cache is updated. This is typically relevant
+     * to Pools that incur protocol debt between operations. To avoid altering the amount due retroactively, this debt
+     * needs to be paid before the fee percentages change.
+     */
+    function _beforeProtocolFeeCacheUpdate() internal virtual {}
+
```

Comment:
Note: all these changes were already included in the base of https://github.com/balancer-labs/balancer-v2-monorepo/pull/2059.